egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -368,11 +370,15 @@ def CreateReverberatedCopy(input_dir,
                            max_noises_per_minute  # maximum number of point-source noises that can be added to a recording according to its duration
                            ):
     
+    wav_scp = ParseFileToDict(input_dir + "/wav.scp", value_processor = lambda x: " ".join(x))
     if not os.path.isfile(input_dir + "/reco2dur"):
         print("Getting the duration of the recordings...");
-        data_lib.RunKaldiCommand("wav-to-duration --read-entire-file=true scp:{0}/wav.scp ark,t:{0}/reco2dur".format(input_dir))
+        read_entire_file="false"
```

Comment:
You are neglecting certain possibilities here:
  - What if some values use sox and some don't?
  - what if sox uses harmless commands that don't affect the header (basically, commands other than --speed)?
Look at the script utils/data/get_utt2dur.sh for inspiration.
Also, dicts in python have a values() function which may be useful here.